### ISSUE TITLE
Copy-pasting removes formatting in rich text editor

### ISSUE THREAD

**Issue by saarantras** [laurent22/joplin#5172]
Like the title says: copy-pasting removes formatting in the rich text editor.

## Environment

Joplin version: 2.1.8
Platform: Linux (Desktop)

## Steps to reproduce
(In the rich text editor)
1. Bold some text
2. Copy it
2. Navigate to a section that is unbolded 
3. Paste the text
4. It is not bolded

## Describe what you expected to happen
I expected the copied text to carry the formatting with it. This is not such a major inconvenience in the provided example, but if you want to, say, paste a long bulleted list with multiple levels of indentation, this is rather frustrating.
I would expect the usual "ctrl-V pastes with formatting, ctrl-shift-V pastes without formatting". Confusingly, ctrl-shift-V simply pastes the unformatted text twice.

**Comment by laurent22** [comment]
Hmm, yes I don't expect that would be fixed or even can be fixed. That's just how Rich Text editors work - when you copy, if your selection doesn't include the (invisible) bold tags then, when you paste, the bold formatting will not be there. The solution is to select a bit before and a bit after the word.

**Comment by saarantras** [comment]
If I select a bit before & after the word then the paste has the tags, but not the formatting: for example, writing **foo**, then selecting a bit before and after, copying and pasting, produces \*\*foo\*\*

**Comment by leematos** [comment]
@saarantras I tested this just now and can confirm, it feels "touchy". In this first test, I think I missed the close for the bold which is why it actually started rendering strange in MD.

https://user-images.githubusercontent.com/359140/125181487-2f216980-e1d3-11eb-9d4e-ce321d363f53.mov


https://user-images.githubusercontent.com/359140/125181420-938ff900-e1d2-11eb-9db3-77faf5cac1d4.mov

that said, if you have "it all" it works fine. For example after this, I tested double clicking on the **bold** text and pasting, and it worked consistently.

**Comment by saarantras** [comment]
"Touchy" is a good word. What about bullet points can you copy & paste them? Even selecting a whole word before & after doesn't seem to work for me..?

**Comment by srijit2002** [comment]
> Hmm, yes I don't expect that would be fixed or even can be fixed. That's just how Rich Text editors work - when you copy, if your selection doesn't include the (invisible) bold tags then, when you paste, the bold formatting will not be there. The solution is to select a bit before and a bit after the word.

Yeah you are right but I think that is very bad for user experience. Like users have to copy a bit before and after everytime and it will be very annoying for them.

**Comment by saarantras** [comment]
This is still an issue as of joplin 2.3.3
